assets/js/modules/adsense/components/settings/AdBlockingRecoveryApp/index.js

```diff
@@ -0,0 +1,223 @@
+	useBreakpoint,
+} from '../../../../../hooks/useBreakpoint';
+import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED,
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+	MODULES_ADSENSE,
+} from '../../../datastore/constants';
+import { CORE_UI } from '../../../../../googlesitekit/datastore/ui/constants';
+import { CORE_LOCATION } from '../../../../../googlesitekit/datastore/location/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function AdBlockingRecoveryApp() {
+	const breakpoint = useBreakpoint();
+
+	const settingsURL = useSelect(
+		( select ) =>
+			`${ select( CORE_SITE ).getAdminURL(
+				'googlesitekit-settings'
+			) }#/connected-services/adsense`
+	);
+	const initialActiveStep = useSelect( ( select ) => {
+		const adBlockingRecoverySetupStatus =
+			select( MODULES_ADSENSE ).getAdBlockingRecoverySetupStatus();
+
+		switch ( adBlockingRecoverySetupStatus ) {
+			case '':
+				return 0;
+			case AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED:
+				return 1;
+			case AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED:
+				return 2;
+		}
+	} );
+	const createMessageCTAClicked = useSelect(
+		( select ) =>
+			!! select( CORE_UI ).getValue(
+				AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED
+			)
+	);
+	const dashboardURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' )
+	);
+
+	const {
+		saveSettings,
+		setAdBlockingRecoverySetupStatus,
+		setUseAdBlockerDetectionSnippet,
+		setUseAdBlockerDetectionErrorSnippet,
+	} = useDispatch( MODULES_ADSENSE );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	const [ activeStep, setActiveStep ] = useState( initialActiveStep );
+
+	const onCancel = useCallback( async () => {
+		if ( activeStep === 0 ) {
+			return navigateTo( dashboardURL );
+		}
+
+		if ( createMessageCTAClicked ) {
+			return navigateTo( settingsURL );
+		}
+
+		setAdBlockingRecoverySetupStatus( '' );
+		setUseAdBlockerDetectionSnippet( false );
+		setUseAdBlockerDetectionErrorSnippet( false );
+
+		const { error } = await saveSettings();
+
+		if ( ! error ) {
+			navigateTo( dashboardURL );
+		}
+	}, [
+		activeStep,
+		createMessageCTAClicked,
+		dashboardURL,
+		navigateTo,
+		saveSettings,
+		setAdBlockingRecoverySetupStatus,
+		setUseAdBlockerDetectionErrorSnippet,
+		setUseAdBlockerDetectionSnippet,
+		settingsURL,
+	] );
+
+	useEffect( () => {
+		if ( undefined === activeStep && undefined !== initialActiveStep ) {
+			setActiveStep( initialActiveStep );
+		}
+	}, [ activeStep, initialActiveStep ] );
+
+	const isTabletWidthOrLarger = breakpoint !== BREAKPOINT_SMALL;
+
+	return (
+		<Fragment>
+			<Header>
+				<HelpMenu />
+			</Header>
+			<div className="googlesitekit-ad-blocking-recovery googlesitekit-module-page">
+				<Grid>
+					<Row>
+						<Cell size={ 12 }>
+							<Layout rounded>
```

Comment:
This isn't an overly-large component on the whole, but this `Layout` and within seems like a good place to extract one which would leave the top-level index here as the skeleton for the page. Maybe we'd call this the `SetupMain` of the app?

Reply:
I agree. Updated the structure accordingly, thanks!